### PR TITLE
feat(table): Add computation of iceberg stats from parquet files

### PR DESCRIPTION
**Pull request by zeroshade**
This one's a big one!

Adding initial implementation of deriving and computing the Iceberg metadata statistics from Parquet files in order to facilitate adding files directly. This includes utilizing the Parquet file metadata retrieve the typed statistics and collecting them based on optional metrics modes and so on. This also includes various unit tests to confirm it is working.

### CODE REVIEW

File: table/arrow_utils.go [hunk 1]
```diff
@@ -892,3 +899,356 @@ func ToRequestedSchema(ctx context.Context, requested, fileSchema *iceberg.Schem
...
+	switch dt.(type) {
+	case iceberg.StringType:
+	case iceberg.BinaryType:
+	default:
+		if metMode.typ == metricModeTruncate {
+			metMode = metricsMode{typ: metricModeFull, len: 0}
+		}
+	}
+
+	isNested := strings.Contains(colName, ".")
```
Comment (kevinjqliu) [inline table/arrow_utils.go:1036]: this might be an issue for columns with `.` in its name

**Review by kevinjqliu — APPROVED** [review]
LGTM 
btw, in the pyiceberg implementation, there are still a few edge cases for stat collections; for example, when running stats collection for the nyc taxi dataset. It would be a good litmus test to try that out here too

File: table/arrow_utils.go (hunk 1 above)
Comment (zeroshade) [reply table/arrow_utils.go:1036]: Columns with `.` in their name would prove kinda difficult for a lot of parquet readers since Parquet uses `.` as the path separator for determining full paths to columns/nodes. 

Also, I got this from pyiceberg :smile: https://github.com/apache/iceberg-python/blob/main/pyiceberg/io/pyarrow.py#L2085
